src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs

```diff
@@ -22,16 +23,36 @@ public static class HalLearnersCatalog
         /// <param name="labelColumn">The labelColumn column.</param>
         /// <param name="featureColumn">The features column.</param>
         /// <param name="weights">The weights column.</param>
-        /// <param name="advancedSettings">Algorithm advanced settings.</param>
         public static OlsLinearRegressionTrainer OrdinaryLeastSquares(this RegressionContext.RegressionTrainers ctx,
             string labelColumn = DefaultColumnNames.Label,
             string featureColumn = DefaultColumnNames.Features,
-            string weights = null,
-            Action<OlsLinearRegressionTrainer.Arguments> advancedSettings = null)
+            string weights = null)
+        {
+            Contracts.CheckValue(ctx, nameof(ctx));
+            var env = CatalogUtils.GetEnvironment(ctx);
```

Comment:
Why do we do this when the other API needs the same context?

Reply:
the new convention is to have two APIs: one for basic usage that let's you set the column names, and most used params, the other one that takes in Options, and allows to set everything. 

---
In reply to: [248767099](https://github.com/dotnet/machinelearning/pull/2170#discussion_r248767099) [](ancestors = 248767099)

Reply:
Oh, I meant, why do we generate a new `env` when the other API needs the same `context` as we have in `ctx`?

Reply:
the trainer ctor needs an IHostEnvironment. 
The 
`
CatalogUtils.GetEnvironment(ctx)
`

does not generate a new environment, just gets the one from within the context. 



---
In reply to: [248781925](https://github.com/dotnet/machinelearning/pull/2170#discussion_r248781925) [](ancestors = 248781925)